appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/SenderServiceBinder.kt

```diff
@@ -97,34 +98,38 @@ class SenderServiceBinder(private val context: Context?) {
         }
     }
 
-    fun startSendKinAsnyc(receiverAddress: String, amount: Int, memo: String, callback: KinTransferCallback) {
+    fun startSendKinAsync(senderAppName: String, senderAppId: String, receiverAddress: String, amount: Int, memo: String, callback: KinTransferCallback) {
         if (isBounded) {
             executorService.execute {
-                transferService?.transferKinAsync(receiverAddress, amount, memo, callback)
+                transferService?.transferKinAsync(senderAppName, senderAppId, receiverAddress, amount, memo, callback)
             }
         }
     }
 
     @Throws(ServiceConfigurationException::class)
     fun bind() {
         if (!isBounded) {
-            val intent = Intent()
-            val senderPackageName = context?.packageName
-            val serviceFullPath = "$senderPackageName.${Consts.SERVICE_DEFAULT_PACKAGE}.${Consts.SENDER_SERVICE_NAME}"
-            intent.component = ComponentName(senderPackageName, serviceFullPath)
-            intent.`package` = senderPackageName
-            context?.packageManager?.let {
-                val resolveInfos: MutableList<ResolveInfo> = it.queryIntentServices(intent, 0)
-                if (!resolveInfos.any()) {
-                    throw ServiceConfigurationException(serviceFullPath, "Service not found - Service must be implemented")
-                }
-                if (resolveInfos.filter { it.serviceInfo.exported }.any()) {
-                    throw ServiceConfigurationException(serviceFullPath, "Service should not be exported")
+            context?.let {
```

Comment:
why did you add this check if  - its null there is no - else then... some error

Reply:
there was no else before either ;-) 
Anyway this is called at onResume - changes of the context being null are zero.
I needed to pass a non-optional to the getInstance method